src/meta/processors/admin/HBProcessor.cpp

```diff
@@ -39,16 +39,39 @@ void HBProcessor::process(const cpp2::HBReq& req) {
     }
 
     LOG(INFO) << "Receive heartbeat from " << host;
-    HostInfo info(time::WallClock::fastNowInMilliSec());
-    auto ret = ActiveHostsMan::updateHostInfo(kvstore_, host, info);
+    std::vector<kvstore::KV> data;
+    auto ret = kvstore::ResultCode::SUCCEEDED;
+    if (req.host.ip != -1 && req.host.port != -1) {
+        TimeInfo hbInfo(time::WallClock::fastNowInMilliSec());
+        ret = ActiveHostsMan::updateHostInfo(this->kvstore_, host, hbInfo);
+
+        if (req.__isset.leader_parts) {
+            nebula::cpp2::HostAddr storageNode;
+            storageNode.set_ip(req.host.ip);
+            storageNode.set_port(req.host.port);
+            const auto& leaderIds = std::move(req).get_leader_parts();
+            std::for_each(leaderIds->begin(), leaderIds->end(), [&] (auto& lp) {
+                data.emplace_back(MetaServiceUtils::leaderKey(storageNode, lp.first),
+                                  MetaServiceUtils::leaderVal(lp.second));
+            });
+        }
+    }
     resp_.set_code(to(ret));
     if (ret == kvstore::ResultCode::ERR_LEADER_CHANGED) {
         auto leaderRet = kvstore_->partLeader(kDefaultSpaceId, kDefaultPartId);
         if (nebula::ok(leaderRet)) {
             resp_.set_leader(toThriftHost(nebula::value(leaderRet)));
         }
     }
-    onFinished();
+
+    int64_t lastUpdateTime = LastUpdateTimeMan::get(this->kvstore_);
+    resp_.set_last_update_time_in_ms(lastUpdateTime);
+
+    if (!data.empty()) {
+        doPut(std::move(data), false);
```

Comment:
Why don't we refresh `last_update_time` here?

Reply:
The `last_update_time` is used to record some changes(insert, update, delete) that require meta client synchronization(eg. config, schema, parts map).  Here the `doPut` operation only update the storaged host time info(keep it online)  by heartbeat. The information does not  need to sync by metaclient.

Reply:
👍